AssertionError: insufficient data in buffer: requested 4 remaining 0

Here is the error i got.
GelDB version `7.0-dev.9399+e5e6e72` 

I am trying to insert array of tuples.

part of schema: `pairs: array<tuple<pair_id: int32, quote_idx: int32, quote_ca: str, quote_token: str, is_canonical: bool>>;`

```
Error inserting token K: Error(
    Inner {
        code: 16777216,
        messages: [
            "AssertionError: insufficient data in buffer: requested 4 remaining 0",
        ],
        error: None,
        headers: {
            1: b"This is most likely a bug in Gel. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md",
            257: b"Traceback (most recent call last):\n  File \"edb/server/protocol/binary.pyx\", line 1069, in edb.server.protocol.binary.EdgeConnection.main_step\n  File \"edb/server/protocol/binary.pyx\", line 1020, in execute\n  File \"edb/server/protocol/binary.pyx\", line 763, in _execute\n  File \"edb/server/protocol/binary.pyx\", line 764, in edb.server.protocol.binary.EdgeConnection._execute\n  File \"edb/server/protocol/execute.pyx\", line 410, in execute\n  File \"edb/server/protocol/execute.pyx\", line 299, in edb.server.protocol.execute.execute\n  File \"edb/server/protocol/args_ser.pyx\", line 207, in edb.server.protocol.args_ser.recode_bind_args\n  File \"edb/server/protocol/args_ser.pyx\", line 512, in edb.server.protocol.args_ser._decode_tuple_args\n  File \"edb/server/protocol/args_ser.pyx\", line 448, in edb.server.protocol.args_ser._decode_tuple_args_core\n  File \"edb/server/pgproto/./frb.pxd\", line 27, in edb.server.pgproto.pgproto.frb_read\n  File \"edb/server/pgproto/./frb.pyx\", line 10, in edb.server.pgproto.pgproto.frb_check\nedb.errors.InternalServerError: AssertionError: insufficient data in buffer: requested 4 remaining 0\n",
        },
        fields: {
            (
                "source_code",
                TypeId(0x94ff3f0e5737f34b0effa8a39e1851a9),
            ): Any { .. },
            (
                "capabilities",
                TypeId(0x86474051b9d758be698236851a2bd44e),
            ): Any { .. },
        },
    },
)
```

--- a/edb/server/protocol/args_ser.pyx
+++ b/edb/server/protocol/args_ser.pyx
@@ -471,10 +471,13 @@ cdef _decode_tuple_args_core(
             raise errors.InputDataError("unsupported array dimensions")
         frb_read(&sub_buf, 4)  # flags
         frb_read(&sub_buf, 4)  # reserved
-        cnt = <uint32_t>hton.unpack_int32(frb_read(&sub_buf, 4))
-        val = hton.unpack_int32(frb_read(&sub_buf, 4)) # bound
-        if val != 1:
-            raise errors.InputDataError("unsupported array bound")
+        if val == 0:
+            cnt = 0
+        else:
+            cnt = <uint32_t>hton.unpack_int32(frb_read(&sub_buf, 4))
+            val = hton.unpack_int32(frb_read(&sub_buf, 4)) # bound
+            if val != 1:
+                raise errors.InputDataError("unsupported array bound")
 
         # For nested arrays, we need to produce an array containing
         # the start/end indexes in the flattened array.

